src/container-mapping.ts

```diff
@@ -158,29 +188,37 @@ export class ContainerMapping implements IMicrosoftSecurityDevOps {
                 timeout: 2500,
                 headers: {
                     'Content-Type': 'application/json',
-                    'Authorization': 'Bearer ' + bearerToken
-                },
-                data: data
+                    'Authorization': 'Bearer ' + bearerToken,
+                    'Content-Length': data.length
+                }
             };
             core.debug(`${options['method'].toUpperCase()} ${url}`);
-    
+
             const req = https.request(url, options, (res) => {
                 let resData = '';
                 res.on('data', (chunk) => {
                     resData += chunk.toString();
                 });
-    
+
                 res.on('end', () => {
                     core.debug('API calls finished. Time taken: ' + (new Date().getMilliseconds() - apiTime) + "ms");
-                    core.debug('Response: ' + resData);
+                    core.debug(`Status code: ${res.statusCode} ${res.statusMessage}`);
+                    core.debug('Response headers: ' + JSON.stringify(res.headers));
+                    if (resData.length > 0) {
+                        core.debug('Response: ' + resData);
+                    }
+                    if (res.statusCode < 200 || res.statusCode >= 300) {
+                        reject(`Received Failed Status code when calling url: ${res.statusCode} ${resData}`);
+                    }

```

Comment:
Should add `return;` to prevent `resolve();` from also getting called and entering a race condition. :)